docs/04-reference/wingsdk-spec.md

```diff
@@ -292,22 +292,63 @@ resource Bucket {
 
   /**
    * Get a file from the bucket.
+   *
+   * @throws Will throw if the file doesn't exist.
    */
   inflight get(key: str): Blob;
 
+  /**
+   * Get a file from the bucket if it exists.
+   */
+  inflight try_get(key: str): Blob?;
+
   /**
    * Delete a file from the bucket.
+   *
+   * @throws Will throw if the file doesn't exist.
    */
   inflight delete(key: str): void;
 
+  /**
+   * Delete a file from the bucket if it exists.
+   */
+  inflight try_delete(key: str): bool;
+
+  /**
+   * Check if a file exists in the bucket.
+   */
+  inflight exists(key: str): bool;
+
+  /**
+   * Get the metadata of a file in the bucket.
+   */
+  inflight stats(key: str): FileMetadata;
```

Comment:
I thought `stat`, `stats`, `metadata` could all be good names, I'm curious which sounds best to other peoples' ears?

Reply:
I personally like metadata

Reply:
Hard question.

Coming from programming in linux, [stats](https://linux.die.net/man/2/stat) feels natural for me.
But I don't think it will be the background of the 99% 

From some reason AWS is using head-object api

Reply:
Paging @benkehoe 👋 I think `stat` sends a message that buckets are file systems and that's probably not the [best mental model](https://twitter.com/ben11kehoe/status/1635703838865870850?s=46&t=vQsWtya6IrJ2YERDZL4eYA).

I would go with `metadata` or something from that family.

Reply:
Yes I was also thinking about updating some of the messaging here to switch out references to "files" with "objects"